app/controllers/api/v0/persons_controller.rb

```diff
@@ -15,10 +15,15 @@ def index
   end
 
   def show
-    person = Person.current.includes(:user, :ranksSingle, :ranksAverage).find_by_wca_id!(params[:wca_id])
+    wca_id = params[:wca_id]
+    person = Person.current.includes(:user, :ranksSingle, :ranksAverage).find_by_wca_id!(wca_id)
     private_attributes = []
-    if current_user && current_user.can_admin_results?
-      private_attributes = %w[incorrect_wca_id_claim_count dob]
+    if current_user
+      if current_user&.wca_id == wca_id || current_user&.any_kind_of_delegate?
+        private_attributes = %w[dob]
+      elsif current_user&.can_admin_results?
+        private_attributes = %w[incorrect_wca_id_claim_count dob]
+      end
```

Comment:
Also, more nit (sorry!): You're wrapping this whole block inside an `if current_user` check. So you already know that `current_user` won't be nil, which makes the `&.` completely unnecessary.

Either:
- Get rid of the `&.` and use simple `.`
- or Drop the outer `if  current_user` completely

(both will effectively do the same thing and both will work, because `if nil` is false in Ruby)